### PR TITLE
feat(LIVE-3127): Add cosmos account banner 

### PR DESCRIPTION
**Pull request by sarneijim**
### 📝 Description

Add cosmos account banner
3 modes:
- Delegate
- Redelegate
- Hidden

### ❓ Context

- **Impacted projects**: `` 
   - Mobile and Desktop
- **Linked resource(s)**: `` 
   - JIRA:  https://ledgerhq.atlassian.net/browse/LIVE-3127

### ✅ Checklist

- [x] **Test coverage** 
- [ ] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

Screenshot 2022-08-19 at 13 58 56

### 🚀 Expectations to reach

Scope: 
- Basic util in common
- Implementation in Desktop
- Implementation in Mobile

### CODE REVIEW

File: libs/ledger-live-common/src/families/cosmos/banner.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { getCurrentCosmosPreloadData } from "./preloadedData";
+import { getAccountBridge } from "../../bridge";
+import { LEDGER_VALIDATOR_ADDRESS } from "./utils";
+import type { CosmosAccount } from "./types";
+
+export async function getAccountBannerState(account: CosmosAccount) {
```
Comment (haammar-ledger) [inline libs/ledger-live-common/src/families/cosmos/banner.ts:6]: Can you please add the return type for this function?

File: apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const { display, redelegate, ledgerValidator, validatorSrcAddress } = state;
+
+  const title = redelegate
+    ? t("account.banner.redelegation.title")
+    : t("account.banner.delegation.title");
+  const description = redelegate
+    ? t("account.banner.redelegation.description")
+    : t("account.banner.delegation.description", {
+        asset: "ATOM",
+        commission: ledgerValidator.commission * 100,
```
Comment (haammar-ledger) [inline apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx:14]: While in theory `ledgerValidator` should never be null or undefined here, I would put a fail-safe since unfortunately we can't rely on TS yet.

File: apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ? t("account.banner.redelegation.cta")
+    : t("account.banner.delegation.cta");
+
+  const onClick = () => {
+    if (redelegate) {
+      dispatch(
+        openModal("MODAL_COSMOS_REDELEGATE", {
+          account,
+          validatorAddress: validatorSrcAddress,
+          validatorDstAddress: ledgerValidator.validatorAddress,
```
Comment (haammar-ledger) [inline apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx:26]: same here

File: apps/ledger-live-desktop/src/renderer/families/cosmos/RedelegationFlowModal/Body.jsx [hunk 4]
```diff
@@ -142,7 +142,7 @@ const Body = ({
 
     const transaction = bridge.updateTransaction(t, {
       mode: "redelegate",
-      validators: [{ address: "", amount: source?.amount ?? BigNumber(0) }],
+      validators: [{ address: validatorDstAddress, amount: source?.amount ?? BigNumber(0) }],
```
Comment (haammar-ledger) [inline apps/ledger-live-desktop/src/renderer/families/cosmos/RedelegationFlowModal/Body.jsx:145]: This seems unrelated to the specific scope of the banner, can you please explain this change?

**Review by haammar-ledger — COMMENTED** [review]
Good job!

File: apps/ledger-live-desktop/static/i18n/en/app.json [hunk 5]
```diff
@@ -745,7 +745,19 @@
     "bandwidthTooltip": "Freeze assets to gain bandwidth",
     "energyTooltip": "Freeze assets to gain energy",
     "delegatedTooltip": "Delegated assets are used in the voting process. This represents your total number of votes.",
-    "undelegatingTooltip": "Undelegated assets are in a timelock of {{timelockInDays}} days, before being available."
+    "undelegatingTooltip": "Undelegated assets are in a timelock of {{timelockInDays}} days, before being available.",
+    "banner": {
+      "redelegation": {
+        "title": "In case you didn’t know...",
```
[the remark this answers is not in the document]
Comment (sarneijim) [reply apps/ledger-live-desktop/static/i18n/en/app.json:751]: I am following the last design from Diego.  https://www.figma.com/file/2wT8aWfGO0XyDu0PVGKF8E/LLD-%26-LLM-%2F-Comos-redelegation

I f you want discuss with him the messages, I can create a meeting.

File: apps/ledger-live-desktop/src/renderer/families/cosmos/RedelegationFlowModal/Body.jsx (hunk 4 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/families/cosmos/RedelegationFlowModal/Body.jsx:145]: In the actual relegation flow is not been selected by default any destiny provider, in this case follow the story we are using 'ledger' as default redelegator destiny. (Ledger in this case is preselected)

File: apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx (hunk 3 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx]: Done

File: apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx (hunk 2 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/families/cosmos/utils.jsx]: Done

File: libs/ledger-live-common/src/families/cosmos/banner.ts (hunk 1 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/families/cosmos/banner.ts]: Done

File: apps/ledger-live-desktop/static/i18n/en/app.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (haammar-ledger) [reply apps/ledger-live-desktop/static/i18n/en/app.json:751]: No need, don't want to complicate things :smile: it was just in case the wording was never defined explicitely.

**Review by haammar-ledger — APPROVED** [review]
:rocket:

**Review by valpinkman — CHANGES_REQUESTED** [review]
@sarneijim you need to add changesets for the changes you've made. You can find the documentation on the tool right here. https://github.com/LedgerHQ/ledger-live/wiki/Changesets 

Don't hesitate to ping me if you have any questions

File: libs/ledger-live-common/src/families/cosmos/banner.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    ) {
+      // Not found worst validator than ledger
+      const maxSpendable = await getAccountBridge(account, undefined)
+        .estimateMaxSpendable({
+          account,
+          parentAccount: undefined,
+          transaction: undefined,
+        })
+        .then((estimate) => {
+          return +estimate;
```
[the remark this answers is not in the document]
Comment (sarneijim) [reply libs/ledger-live-common/src/families/cosmos/banner.ts]: Changed thanks!

**Comment by sarneijim** [comment]
> @sarneijim you need to add changesets for the changes you've made. You can find the documentation on the tool right here. https://github.com/LedgerHQ/ledger-live/wiki/Changesets
> 
> Don't hesitate to ping me if you have any questions

Done. I assume as it is not a breaking change is a minor change for the 3 packages. :)

File: .changeset/new-doors-flow.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+"ledger-live-desktop": patch
```
Comment (valpinkman) [inline .changeset/new-doors-flow.md:2]: @sarneijim I think you want to be a minor here.

Also please split into three different changesets (yes it's a pain in the ass due to the release flow right now but hopefully we can work that out soon)

**Review by valpinkman — CHANGES_REQUESTED** [review]

**Comment by valpinkman** [comment]
> > @sarneijim you need to add changesets for the changes you've made. You can find the documentation on the tool right here. https://github.com/LedgerHQ/ledger-live/wiki/Changesets
> > Don't hesitate to ping me if you have any questions
> 
> Done. I assume as it is not a breaking change is a minor change for the 3 packages. :)

yes you are right it's suppose to me minor (yet you've used patch i think)

File: .changeset/quiet-singers-beg.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"live-mobile": major
```
Comment (valpinkman) [inline .changeset/quiet-singers-beg.md:2]:
```suggestion
"live-mobile": minor
```

File: .changeset/small-donkeys-hope.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+"ledger-live-desktop": major
```
Comment (valpinkman) [inline .changeset/small-donkeys-hope.md:2]:
```suggestion
"ledger-live-desktop": minor
```

File: .changeset/tough-geckos-reflect.md [hunk 10]
```diff
@@ -0,0 +1,5 @@
+---
+"@ledgerhq/live-common": major
```
Comment (valpinkman) [inline .changeset/tough-geckos-reflect.md:2]:
```suggestion
"@ledgerhq/live-common": minor
```

**Review by valpinkman — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/renderer/screens/account/index.jsx [hunk 11]
```diff
@@ -39,6 +39,10 @@ import EmptyStateAccount from "./EmptyStateAccount";
 import TokensList from "./TokensList";
 import CompoundBodyHeader from "~/renderer/screens/lend/Account/AccountBodyHeader";
 import useCompoundAccountEnabled from "~/renderer/screens/lend/useCompoundAccountEnabled";
+import { getBannerProps, AccountBanner } from "./AccountBanner";
+
+import { useDispatch } from "react-redux";
```
Comment (valpinkman) [inline apps/ledger-live-desktop/src/renderer/screens/account/index.jsx:44]: those two imports use libraries that are already called a bit earlier in the file. Could you group them ?
`react-redux` and `react-i18next`

**Review by valpinkman — CHANGES_REQUESTED** [review]

File: apps/ledger-live-desktop/src/renderer/screens/account/index.jsx (hunk 11 above)
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/screens/account/index.jsx]: Done

**Review by valpinkman — APPROVED** [review]
